protocol/contact.go

```diff
@@ -148,27 +105,20 @@ func BuildContactFromPublicKey(publicKey *ecdsa.PublicKey) (*Contact, error) {
 }
 
 func buildContact(publicKeyString string, publicKey *ecdsa.PublicKey) (*Contact, error) {
-	identicon, err := identicon.GenerateBase64(publicKeyString)
+	newIdenticon, err := identicon.GenerateBase64(publicKeyString)
 	if err != nil {
 		return nil, err
 	}
 
 	contact := &Contact{
 		ID:        publicKeyString,
 		Alias:     alias.GenerateFromPublicKey(publicKey),
-		Identicon: identicon,
+		Identicon: newIdenticon,
 	}
 
 	return contact, nil
 }
```

Comment:
If we are looking for optimisation, we could offload some computation work by storing the identicon payload in the DB and retrieving it rather than building it every time we want access to the contact.

This would come at the cost of an increased DB size, but an identicon never changes so it is the perfect thing to store.

Reply:
yeah, we're thinking about that, but will be addressed in another PR